src/zenml/zen_stores/secrets_stores/gcp_secrets_store.py

```diff
@@ -0,0 +1,612 @@
+            )
+            updated = datetime.strptime(
+                label_dict[ZENML_GCP_SECRET_UPDATED_KEY],
+                ZENML_GCP_DATE_FORMAT_STRING,
+            )
+        except KeyError as e:
+            raise KeyError(
+                f"Invalid GCP secret: missing required tag '{e}'"
+            ) from e
+
+        return self._create_secret_from_metadata(
+            metadata=label_dict,
+            created=created,
+            updated=updated,
+            values=values,
+        )
+
+    def _get_gcp_filter_string(
+        self, secret_filter_model: SecretFilterModel
+    ) -> str:
+        """Convert a SecretFilterModel to a GCP filter string.
+
+        Args:
+            secret_filter_model: The secret filter model.
+
+        Returns:
+            The GCP filter string.
+        """
+        operator_map = {
+            "equals": ":",
+        }
+        filter_terms = []
+        for filter in secret_filter_model.list_of_filters:
+            filter_terms.append(
+                f"{filter.column}{operator_map[filter.operation.value]}{filter.value}"
+            )
+
+        return f" {secret_filter_model.logical_operator.name} ".join(
+            filter_terms
+        )
+
+    @track(AnalyticsEvent.CREATED_SECRET)
+    def create_secret(self, secret: SecretRequestModel) -> SecretResponseModel:
+        """Create a new secret.
+
+        The new secret is also validated against the scoping rules enforced in
+        the secrets store:
+
+          - only one workspace-scoped secret with the given name can exist
+            in the target workspace.
+          - only one user-scoped secret with the given name can exist in the
+            target workspace for the target user.
+
+        Args:
+            secret: The secret to create.
+
+        Returns:
+            The created secret.
+
+        Raises:
+            RuntimeError: if the secret was unable to be created.
+            EntityExistsError: If a secret with the same name already exists
+                in the same scope.
+        """
+        self._validate_gcp_secret_name(secret.name)
+
+        user, workspace = self._validate_user_and_workspace(
+            secret.user, secret.workspace
+        )
+
+        # Check if a secret with the same name already exists in the same
+        # scope.
+        secret_exists, msg = self._check_secret_scope(
+            secret_name=secret.name,
+            scope=secret.scope,
+            workspace=secret.workspace,
+            user=secret.user,
+        )
+        if secret_exists:
+            raise EntityExistsError(msg)
+
+        secret_id = uuid.uuid4()
+        secret_value = json.dumps(secret.secret_values)
+
+        created = datetime.now(timezone.utc).replace(
+            tzinfo=None, microsecond=0
+        )
+        labels = self._get_secret_metadata_for_secret(
+            secret=secret, secret_id=secret_id
+        )
+        labels[ZENML_GCP_SECRET_CREATED_KEY] = created.strftime(
```

Comment:
yes, GCP has extra restrictions around label values: we can't use any uppercase letters or semicolons in the value. We had to settle for a custom format because of this.